cmd/kueue/main.go

```diff
@@ -292,7 +293,12 @@ func setupProbeEndpoints(mgr ctrl.Manager) {
 	// the function, otherwise a not fully-initialized webhook server (without
 	// ready certs) fails the start of the manager.
 	if err := mgr.AddReadyzCheck("readyz", func(req *http.Request) error {
-		return mgr.GetWebhookServer().StartedChecker()(req)
+		select {
+		case <-certsReady:
+			return mgr.GetWebhookServer().StartedChecker()(req)
+		case <-req.Context().Done():
```

Comment:
Maybe use `default` for certs not ready as in the other project:  https://github.com/kubernetes-sigs/hierarchical-namespaces/blob/d367fc08a261135b63d22aeb01c688317d9b7e02/cmd/manager/main.go#L296-L307. For context done we probably can have another dedicated message to distinguish like closing the readiness probe ad the context is closed?

Reply:
It depends if we want to keep the caller busy or not. 

/cc @alculquicondor

Reply:
I think it does not matter because at this point the only caller should be the kubelet readiness check probably, so I would prefer simplicity.

Reply:
I there a spec on if we should block or not?

Reply:
From https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
```
Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.
```
So IIUC we are ok to close as long saying "not ok"

Reply:
we should close the connection ASAP, so let's use default.

Reply:
done